eng/SnippetGenerator/CSharpProcessor.cs

```diff
@@ -11,39 +11,54 @@ namespace SnippetGenerator
 
         private static readonly Regex _snippetRegex = new Regex("^(?<indent>\\s*)\\/{3}\\s*<code snippet=\"(?<name>[\\w:]+)\">.*?\\s*<\\/code>",
             RegexOptions.Compiled | RegexOptions.Multiline | RegexOptions.Singleline | RegexOptions.IgnoreCase);
 
-        private static readonly Regex _snippetExampleRegex = new Regex("^(?<indent>\\s*)\\/{3}\\s*<example snippet=\"(?<name>[\\w:]+)\">.*?\\s*<\\/example>",
+        private static readonly Regex _snippetExampleRegex = new Regex("^(?<indent>\\s*)\\/{3}\\s*<example snippet=\"(?<name>[\\w:]+)\"?>(?<body>.*?\\s*)(^\\s*\\/{3}\\s*)?<\\/example>",
+            RegexOptions.Compiled | RegexOptions.Multiline | RegexOptions.Singleline | RegexOptions.IgnoreCase);
+
+        private static readonly Regex _CodeDocRegex = new Regex("^(?<indent>\\s*)\\/{3}\\s*<code *>.*?\\s*<\\/code>",
             RegexOptions.Compiled | RegexOptions.Multiline | RegexOptions.Singleline | RegexOptions.IgnoreCase);
 
         public static string Process(string markdown, Func<string, string> snippetProvider)
         {
             string CodeTagFormatter(Match match)
             {
-                var name = BuildResult(snippetProvider, match, out var prefix, out var builder);
+                var name = BuildResult(snippetProvider, match, out var prefix, out var builder, out _);
 
                 return string.Format(_snippetFormat, name, Environment.NewLine, builder, prefix);
             }
 
             string ExampleTagFormatter(Match match)
             {
-                var name = BuildResult(snippetProvider, match, out var prefix, out var builder);
+                var name = BuildResult(snippetProvider, match, out var prefix, out var builder, out var exampleBody);
 
-                return string.Format(_snippetExampleFormat, name, Environment.NewLine, builder, prefix);
+                return string.Format(_snippetExampleFormat, name, Environment.NewLine, builder, prefix, exampleBody);
             }
 
             string result = _snippetRegex.Replace(markdown, CodeTagFormatter);
-            return result != markdown ? result : _snippetExampleRegex.Replace(markdown, ExampleTagFormatter);
+            return _snippetExampleRegex.Replace(result, ExampleTagFormatter);
         }
 
-        private static string BuildResult(Func<string, string> snippetProvider, Match match, out string prefix, out StringBuilder builder)
+        private static string BuildResult(Func<string, string> snippetProvider, Match match, out string prefix, out StringBuilder builder, out string exampleBody)
         {
             var name = match.Groups["name"].Value;
             prefix = match.Groups["indent"].Value + "///";
 
+            // set exampleBody to the contents of the body of the tag minus the <code>...</code> block.
+            // this is to preserve any example text prefix to the code block.
+            if (match.Groups["body"].Success)
```

Comment:
We shouldn't do this. If you need to stuff inside the example you can use `<code snippet`.